pjlib/src/pj/ioqueue_epoll.c

```diff
@@ -912,26 +1045,20 @@ PJ_DEF(int) pj_ioqueue_poll( pj_ioqueue_t *ioqueue, const pj_time_val *timeout)
     /* Special case:
      * When epoll returns > 0 but event_cnt, the number of events
      * we want to process, is zero.
-     * There are several possibilities this can happen:
-     * - Multiple polling threads can receive the same event
-     *   (if not EXCLUSIVE nor ONESHOT), and only one thread will
-     *   process it.
-     * - Events EPOLLIN is always on, while EPOLLERR and EPOLLHUP
-     *   will always be automatically reported even though we
-     *   never specifically asked for them.
+     * There are several possibilities this can happen, see "UNHANDLED event"
+     * log message above.
      */
-    if (count > 0 && !event_cnt && msec > 0 &&
-    	!ioqueue->use_epollexlusive && !ioqueue->use_epolloneshot)
-    {
+    if (count > 0 && !event_cnt && msec > 0) {
 	/* We need to sleep in order to avoid busy polling, such
          * as in the case of the thread that doesn't process
          * the event as explained above.
-         * Note that the sleep period should be reduced by
-         * the amount of time already used for epoll_wait().
+         * Limit the duration of the sleep, as doing pj_thread_sleep() for
```

Comment:
First of all, the sleep here is to compensate/prevent busy loop for the case when epoll is woken up due to "stray" events. I don't think there is equivalent case in select() ioqueue? There is a different case of sleep in select() ioqueue (`ioqueue_select.c:989`), but i don't want to fix that for now since it's not part of the problem?